sdk/eventhub/Azure.Messaging.EventHubs/src/EventData.cs

```diff
@@ -50,7 +57,18 @@ public class EventData
         ///   </code>
         /// </example>
         ///
-        public IDictionary<string, object> Properties { get; }
+        public IDictionary<string, object> Properties
+        {
+            get
+            {
+                if (_properties == null)
+                {
+                    Interlocked.CompareExchange(ref _properties, new Dictionary<string, object>(), null);
```

Comment:
I'm not sure if the `Interlocked` use is overkill, since we normally don't guarantee thread safety for models.  An alternate could be something like:

```csharp
public IDictionary<string, object> Properties
{
    get => _properties ??= new Dictionary<string, object>();
}
    
public IReadOnlyDictionary<string, object> SystemProperties 
{ 
    get => _systemProperties ??= new Dictionary<string, object>();
}
```